### PR TITLE
🌱 update to CAPI 1.7.2

### PR DESCRIPTION
**Pull request by kranurag7**
**What this PR does / why we need it**:
updates CAPI to 1.7.2

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits
- [ ] include documentation
- [ ] add unit tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -132,13 +132,13 @@ $(HELM):
...
 $(HCLOUD):
-	curl -sSL https://github.com/hetznercloud/cli/releases/download/v1.42.0/hcloud-linux-amd64.tar.gz | tar xz -C $(TOOLS_BIN_DIR) hcloud
+	curl -sSL https://github.com/hetznercloud/cli/releases/download/v1.43.1/hcloud-linux-amd64.tar.gz | tar xz -C $(TOOLS_BIN_DIR) hcloud
 	chmod a+rx $(HCLOUD)
 
 KIND := $(abspath $(TOOLS_BIN_DIR)/kind)
 kind: $(KIND) ## Build a local copy of kind
 $(KIND):
-	go install sigs.k8s.io/kind@v0.22.0
+	go install sigs.k8s.io/kind@v0.23.0
```
Comment (janiskemper) [inline Makefile:141]: does CAPI bump kind as well? I thought they usually do that with every minor version

**Review by janiskemper — CHANGES_REQUESTED** [review]
you forgot the docs. Just search for "1.7" there.

File: Makefile (hunk 1 above)
Comment (kranurag7) [reply Makefile:141]: @janiskemper this has nothing to do with go bump of CAPI. This is for installing latest version of kind (binary) in hack/tools/bin directory.

Comment (janiskemper) [reply Makefile:141]: what about this then? https://cluster-api.sigs.k8s.io/developer/providers/migrations/v1.6-to-v1.7

Comment (kranurag7) [reply Makefile:141]: CAPI updated it here https://github.com/kubernetes-sigs/cluster-api/pull/10610 

and we have an open PR for the same here https://github.com/syself/cluster-api-provider-hetzner/pull/1308

Comment (janiskemper) [reply Makefile:141]: they will launch it in 1.8. But since this is no go module, it might be less important to wait.

**Review by janiskemper — APPROVED** [review]

**Comment by rbjorklin** [comment]
Should the [compatibility section of the README](https://github.com/syself/cluster-api-provider-hetzner?tab=readme-ov-file#fire-compatibility-with-cluster-api-and-kubernetes-versions) have been updated along with this PR?

**Comment by kranurag7** [comment]
> Should the [compatibility section of the README](https://github.com/syself/cluster-api-provider-hetzner?tab=readme-ov-file#fire-compatibility-with-cluster-api-and-kubernetes-versions) have been updated along with this PR?

This was in discussion to update it after this PR was merged but it was me who missed it. I'll create a PR updating the same.

**Pull request merged**